src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Services/ExtensionAssemblyManagerProvider.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Services;
+
+/// <summary>
+/// Provider to allow MEF access to <see cref="ExtensionAssemblyManager"/>
+/// Must be done this way as the manager is required to create MEF as well.
+/// </summary>
```

Comment:
This is fine to go with for now, but I'm guessing there's some way we can directly inject the ExtensionAssemblyManager into the catalog. We do similar things though for logging as it is, so this isn't new.

Reply:
I did take a quick look.  There's nothing in the discovery / catalog APIs that I could find where we could provide a specific instance of an export (we can manually add a type).  This https://github.com/microsoft/vs-mef/blob/main/doc/dynamic_recomposition.md#adding-parts-directly-with-compositioncontainercompose also recommends basically what we're doing.